src/index.ts

```diff
@@ -21,6 +22,89 @@ function writeLoaders(args: CLIArgs) {
     fs.writeFileSync(args.output, output);
 }
 
+/**
+ * If requiredList only contains responseKey
+ */
+function validRequiredList(requiredList: Array<string>, responseKey: string) {
+    if (requiredList.length > 1) return false;
+    if (requiredList.length == 1 && requiredList[0] !== responseKey) {
+        return false;
+    }
+    return true;
+}
+
+/**
+ * Find all values of the intput key in an object recursively
+ */
+function findVal(object: any, key: string, array: Array<string>) {
+    Object.keys(object).some(function (k) {
+        if (k === key) {
+            array.push(...object[k]);
```

Comment:
does eslint not complain at you? we're mutating the input array variable here, which probably isn't what you want (unintended side effects)

objects are pass by reference in js https://www.javascripttutorial.net/javascript-pass-by-value/